roc/version_test.go

```diff
@@ -0,0 +1,16 @@
+package roc
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestVersion(t *testing.T) {
```

Comment:
Great test, although it does only test a successful path. I can see that the way `Version` func is implemented you cannot really inject any test strings into it and test failures here. Have you thought about decoupling it to make it possible? There are a lot of moving parts in `Version` and it can crap out in e. g. lines 40, 47, 51, 52 and 53. I am not sure if it is vital, but I would suggest it.

Reply:
> Have you thought about decoupling it to make it possible? 

yes already thought about this, but seems ok to me to ignore and try to return as much as possible instead of returning error

> There are a lot of moving parts in Version and it can crap out in e. g. lines 40, 47, 51, 52 and 53. I am not sure if it is vital, but I would suggest it.

yes those lines can fail when parsing invalid `bindingsVersion`
let me try add 1 unit test to check whether the `bindingsVersion` is valid or not so we can check it early in unit test

Reply:
> yes already thought about this, but seems ok to me to ignore and try to return as much as possible instead of returning error

In general, I would not think it's ok to do it, but I can see your motivation, because `Version` returns {0,0,0} in case of error anyway. Let's see how that discussion plays out, so depending on what we decide, you would leave this func as is or rewrite it a bit to decouple getting the version from C and parsing it to allow for more modularity and testability.

Reply:
>  let me try add 1 unit test to check whether the bindingsVersion is valid or not so we can check it early in unit test

I like this idea. User interface is simpler (no need to check for errors from roc.Version), and at the same time our test suite guarantees that failures are not possible (by validating hard-coded version).

The only downside is that the test duplicates parsing logic. If we update version.go and forget version_test.go, then the test's validation may become irrelevant.

So I'd suggest to refactor it to reuse parsing code. One way to do it:

1. extract private function parseVersion that accepts string, returns version triple, or panics on invalid string format
2. in roc.Version, use parseVersion
3. in tests, check that:
  - parseVersion works on valid input
  - parseVersion panics on invalid input
  - parseVersion works in hard-coded bindingsVersion

Reply:
+1 from me, @Asalle feel free to resolve this